src/parameters.jl

```diff
@@ -0,0 +1,603 @@
+# Copyright (c) 2020: Akshay Sharma and contributors
+#
+# Use of this source code is governed by an MIT-style license that can be found
+# in the LICENSE.md file or at https://opensource.org/licenses/MIT.
+
+# block other methods
+
+MOI.supports(::POI.Optimizer, ::ForwardObjectiveFunction) = false
+
+function MOI.set(::POI.Optimizer, ::ForwardObjectiveFunction, _)
+    return error(
+        "Forward objective function is not supported when " *
+        "`with_parametric_opt_interface` is set to `true` in " *
+        "`diff_optimizer`." *
+        "Use parameters to set the forward sensitivity.",
+    )
+end
+
+MOI.supports(::POI.Optimizer, ::ForwardConstraintFunction) = false
+
+function MOI.set(
+    ::POI.Optimizer,
+    ::ForwardConstraintFunction,
+    ::MOI.ConstraintIndex,
+    _,
+)
+    return error(
+        "Forward constraint function is not supported when " *
+        "`with_parametric_opt_interface` is set to `true` in " *
+        "`diff_optimizer`." *
+        "Use parameters to set the forward sensitivity.",
+    )
+end
+
+MOI.supports(::POI.Optimizer, ::ReverseObjectiveFunction) = false
+
+function MOI.get(::POI.Optimizer, ::ReverseObjectiveFunction)
+    return error(
+        "Reverse objective function is not supported when " *
+        "`with_parametric_opt_interface` is set to `true` in " *
+        "`diff_optimizer`." *
+        "Use parameters to get the reverse sensitivity.",
+    )
+end
+
+MOI.supports(::POI.Optimizer, ::ReverseConstraintFunction) = false
+
+function MOI.get(
+    ::POI.Optimizer,
+    ::ReverseConstraintFunction,
+    ::MOI.ConstraintIndex,
+)
+    return error(
+        "Reverse constraint function is not supported when " *
```

Comment:
It's not obvious to me why we can't support both

Reply:
I think the worst part would be the user accidentally double-counting stuff for both getters and setters.
In the case of the Forward + set is even more dangerous because setting  through the function could override what was set through parameters and vice versa

Reply:
Yes, erroring is definitely best in the scope of this PR

Reply:
issue created